File: src/CHS/CHS.Entities/Favorite.cs
namespace CHS.Entities
{
    public class Favorite
    {
        #region Fields

        private int favoriteId;
        private User user;
        private Snippet snippet;

        #endregion Fields

        #region Constructors

        public Favorite(int id, User user, Snippet snippet)
        {
            FavoriteId = id;
            User = user;
            Snippet = snippet;
        }

        #endregion Constructors

        #region Properties

        public int FavoriteId
        {
            get { return favoriteId; }
            set { favoriteId = value; }
        }

        public User User
        {
            get { return user; }
            set { user = value; }
        }

        public Snippet Snippet
        {
            get { return snippet; }
            set { snippet = value; }
        }

        #endregion Properties
    }
}

File: src/CHS/CHS.Client/client/src/entities/Favorite.ts
import { User } from "./User";
import { Snippet } from "./Snippet";

export class Favorite {
    private favoriteId: number;
    private user: User;
    private snippet: Snippet;

    constructor(favoriteId: number, user: User, snippet: Snippet) {
        this.favoriteId = favoriteId;
        this.user = user;
        this.snippet = snippet;
    }

    public get FavoriteId(): number {
        return this.favoriteId;
    }

    public set FavoriteId(id: number) {
        this.favoriteId = id;
    }

    public get User(): User {
        return this.user;
    }

    public set User(user: User) {
        this.user = user;
    }

    public get Snippet(): Snippet {
        return this.snippet;
    }

    public set Snippet(snippet: Snippet) {
        this.snippet = snippet;
    }
}
